infoschema/cluster_tables_serial_test.go

```diff
@@ -125,13 +127,15 @@ func SubTestForClusterServerInfo(s *clusterTablesSuite) func(*testing.T) {
 				// Because the underlying implementation use `sysctl` command to get the result
 				// and there is no such command on windows.
 				// https://github.com/pingcap/sysutil/blob/2bfa6dc40bcd4c103bf684fba528ae4279c7ec9f/system_info.go#L50
-				skipOnOS: "windows",
+				skipOnDist: "(windows/.+)",
 			},
 		}
 
 		for _, cas := range cases {
-			if cas.skipOnOS == runtime.GOOS {
-				continue
+			if cas.skipOnDist != "" {
+				if matched, _ := regexp.MatchString(cas.skipOnDist, runtime.GOOS+"/"+runtime.GOARCH); matched {
```

Comment:
replace with `&&`?

Reply:
Ahh, seems not possible to `&&` with a multivalue return?

Reply:
Make sense. I suggest you try to add a utility for detecting specific platforms. Especially for Windows and Apple M1. And thus avoid these inline handmade stuff.

Reply:
It looks quite brittle you bring a new pattern to concatenate from OS + ARCH here and place some regexes above.

Reply:
I agree. I've spent some time trying to figure out a better solution but failed. Maybe use a function to hide the regexes.